packages/salesforcedx-vscode-core/test/jest/commands/deployExecutor.test.ts

```diff
@@ -148,4 +162,140 @@ describe('Deploy Executor', () => {
     expect(ensureLocalTrackingSpy).not.toHaveBeenCalled();
     expect(deploySpy).toHaveBeenCalled();
   });
+
+  it('unsuccessfulOperationHandler', () => {
+    // Arrange
+    const mockDeployResult = {
+      response: {
+        status: 'Failed'
+      }
+    };
+    const handleDeployDiagnosticsSpy = jest
+      .spyOn(diagnostics, 'handleDeployDiagnostics')
+      .mockImplementation(jest.fn());
+    DeployRetrieveExecutor.errorCollection = MockErrorCollection as any;
+    const executor = new TestDeployExecutor(
+      'testDeploy',
+      'force_source_deploy_with_sourcepath_beta'
+    );
+
+    // Act
+    (executor as any).unsuccessfulOperationHandler(
+      mockDeployResult,
+      DeployRetrieveExecutor.errorCollection
+    );
+
+    expect(handleDeployDiagnosticsSpy).toHaveBeenCalledWith(
+      mockDeployResult,
+      DeployRetrieveExecutor.errorCollection
+    );
+  });
+
+  describe('postOperation', () => {
```

Comment:
Separate describe block for postOperation! Good Work on keeping the separation of concerns in mind! 🎉